### PR TITLE
Add unit tests to module

### PR DESCRIPTION
**Pull request by MonssafToukal**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "julia.environmentPath": "/home/monssaf/SolverParameters.jl"
+}
```
Comment (tmigot) [inline .vscode/settings.json]: I don't think we need this.

File: src/parameters.jl [hunk 2]
```diff
@@ -1,76 +1,75 @@
...
-  input_types
+  upper_bounds
+
+
+""" `AbstractParameterSet`
+An abstract type that represents a set of multiple parameters. 
+  Ex:
+  mutable struct MySolverParameterSet <: AbstractParameterSet
+    η₁::Parameter(1.0e-5, RealInterval(0, 1; lower_open=true, upper_open=false))
+  end
```
Comment (tmigot) [inline src/parameters.jl]: Example:
```julia
  mutable struct MySolverParameterSet <: AbstractParameterSet
    η₁::Parameter(1.0e-5, RealInterval(0, 1; lower_open=true, upper_open=false))
  end
```

Comment (tmigot) [reply src/parameters.jl]: Use the "```" delimiters for a block code

File: src/parameters.jl [hunk 3]
```diff
@@ -1,76 +1,75 @@
...
 
 An Abstract type that superseeds the other parameter types.
 """
-abstract type Parameter{T} end
+abstract type AbstractParameter{T} end
 
 """`AlgorithmicParameter`
 
 Child type of `Parameter` that exposes a specific hyperparameter.
 It contains specific fields that can be given to `NOMAD`.
```
Comment (tmigot) [inline src/parameters.jl]:
```suggestion

```

File: test/test_parameters.jl [hunk 4]
```diff
@@ -0,0 +1,6 @@
+@testset "Parameters" verbose=true begin
```
Comment (tmigot) [inline test/test_parameters.jl]: I think we also want to do this tests for all the different types not only `Real`

**Review by tmigot — CHANGES_REQUESTED** [review]
That looks great, thanks! I have a couple of comments to improve.

**Comment by tmigot** [comment]
Any idea why the tests fail with Julia 1.6?

File: src/parameters.jl [hunk 5]
```diff
@@ -1,76 +1,75 @@
...
 """Returns the current value of a parameter."""
-value(parameter::AlgorithmicParameter{T}) where {T} = parameter.value
+value(parameter::Parameter{T}) where {T} = parameter.value
 
-domain(parameter::AlgorithmicParameter{T}) where {T} = parameter.domain
+domain(parameter::Parameter{T}) where {T} = parameter.domain
 
 """Returns the name of a parameter."""
-name(parameter::AlgorithmicParameter{T}) where {T} = parameter.name
+name(parameter::Parameter{T}) where {T} = parameter.name
```
Comment (tmigot) [inline src/parameters.jl]: Are these functions tested?

**Comment by MonssafToukal** [comment]
> Any idea why the tests fail with Julia 1.6?

I think it's because I use regex to check that the correct error message is thrown? I'll change it to a `occursin("error msg", error)` and see if it does the same.

**Comment by MonssafToukal** [comment]
@tmigot Let me know if anything is missing!

**Review by tmigot — APPROVED** [review]
Great job, thanks!

**Pull request merged**